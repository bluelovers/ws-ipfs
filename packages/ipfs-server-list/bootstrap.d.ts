/**
 * Created by user on 2020/3/29.
 */
/**
 * js-ipfs
 * @type {string[]}
 */
export declare const ipfsBootstrapJS: string[];
/**
 * ipfs-desktop
 * @type {string[]}
 */
export declare const ipfsBootstrapDesktop: string[];
export declare const ipfsBootstrapList: string[];
export default ipfsBootstrapList;
