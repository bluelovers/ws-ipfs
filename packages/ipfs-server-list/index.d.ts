import { ITSRequireAtLeastOne, ITSPartialRecord } from 'ts-type';
import { IIPFSClientAddresses } from '@bluelovers/ipfs-http-client';
export declare type ILimit = ITSPartialRecord<'ref' | 'id' | 'config', boolean>;
export declare type IIPFSAddressesLike = ITSRequireAtLeastOne<{
    API?: IIPFSClientAddresses;
    Gateway?: string;
    IPLD?: string;
    IPNS?: string;
    /**
     * https://blog.cloudflare.com/continuing-to-improve-our-ipfs-gateway/
     *
     * base32.cf-ipfs.com
     */
    GatewayDomain?: string;
    IPNSDomain?: string;
    btfsGateway?: string;
}> & {
    /**
     * @deprecated
     */
    limit?: ITSRequireAtLeastOne<ILimit>;
    name?: string;
    description?: string;
};
export declare function getIpfsServerList(): Record<"ipfs" | "infura.io" | "infura-ipfs.io" | "cloudflare" | "cloudflare-ipfs" | "ipfs.gateway" | "dweb" | "fleek" | "bdaily" | "globalupload" | "pinata" | "hardbin" | "eternum" | "temporal" | "sloppyta" | "greyh" | "jorropo" | "jorropo.net" | "jeroendeneef" | "2read" | "runfission" | "best-practice" | "privacytools" | "trusti" | "stibarc" | "dtube" | "dtube.2" | "cosmos-ink" | "storjipfs-gateway" | "permaweb" | "cwinfo" | "fooock" | "serph.network" | "busy.org" | "doolta" | "originprotocol" | "mrh.io" | "ipns.co" | "blocksec" | "10.via0.com" | "ninetailed.ninja" | "geesome-node" | "ipfs.yt" | "overpi" | "adatools.io" | "drink.cafe" | "robotizing.net" | "mihir.ch" | "telos.miami" | "tubby.cloud" | "kaleido.art" | "3cloud.ee" | "crustwebsites.net" | "textile.io" | "itargo.io" | "decoo.io" | "denarius.io" | "jbb.one" | "ravencoinipfs-gateway.com" | "trusted-setup.filecoin.io" | "eth.aragon.network" | "bluelight.link" | "birds-are-nice.me", IIPFSAddressesLike> & Record<string, IIPFSAddressesLike>;
export declare function getIpfsLocalList(): Record<"go-ipfs" | "js-ipfs", IIPFSAddressesLike> & Record<string, IIPFSAddressesLike>;
export declare function filterList<K extends keyof IIPFSAddressesLike>(key: K, serverList?: Record<string, IIPFSAddressesLike>): IIPFSAddressesLike[K][];
export declare const ipfsServerList: Record<"ipfs" | "infura.io" | "infura-ipfs.io" | "cloudflare" | "cloudflare-ipfs" | "ipfs.gateway" | "dweb" | "fleek" | "bdaily" | "globalupload" | "pinata" | "hardbin" | "eternum" | "temporal" | "sloppyta" | "greyh" | "jorropo" | "jorropo.net" | "jeroendeneef" | "2read" | "runfission" | "best-practice" | "privacytools" | "trusti" | "stibarc" | "dtube" | "dtube.2" | "cosmos-ink" | "storjipfs-gateway" | "permaweb" | "cwinfo" | "fooock" | "serph.network" | "busy.org" | "doolta" | "originprotocol" | "mrh.io" | "ipns.co" | "blocksec" | "10.via0.com" | "ninetailed.ninja" | "geesome-node" | "ipfs.yt" | "overpi" | "adatools.io" | "drink.cafe" | "robotizing.net" | "mihir.ch" | "telos.miami" | "tubby.cloud" | "kaleido.art" | "3cloud.ee" | "crustwebsites.net" | "textile.io" | "itargo.io" | "decoo.io" | "denarius.io" | "jbb.one" | "ravencoinipfs-gateway.com" | "trusted-setup.filecoin.io" | "eth.aragon.network" | "bluelight.link" | "birds-are-nice.me", IIPFSAddressesLike> & Record<string, IIPFSAddressesLike>;
export default ipfsServerList;
