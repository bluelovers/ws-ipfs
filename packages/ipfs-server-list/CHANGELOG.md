# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.66](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.65...ipfs-server-list@1.0.66) (2021-07-31)

**Note:** Version bump only for package ipfs-server-list





## [1.0.65](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.64...ipfs-server-list@1.0.65) (2021-07-31)

**Note:** Version bump only for package ipfs-server-list





## [1.0.64](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.63...ipfs-server-list@1.0.64) (2021-07-31)

**Note:** Version bump only for package ipfs-server-list





## [1.0.63](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.62...ipfs-server-list@1.0.63) (2021-07-31)

**Note:** Version bump only for package ipfs-server-list





## [1.0.62](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.61...ipfs-server-list@1.0.62) (2021-07-31)

**Note:** Version bump only for package ipfs-server-list





## [1.0.61](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.60...ipfs-server-list@1.0.61) (2021-07-29)


### ‚ôªÔ∏è„ÄÄChores

* update deps ([2a0dc57](https://github.com/bluelovers/ws-ipfs/commit/2a0dc5779089b83b9a2577995c216cd199c016a5))





## [1.0.60](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.59...ipfs-server-list@1.0.60) (2021-07-23)

**Note:** Version bump only for package ipfs-server-list





## [1.0.59](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.58...ipfs-server-list@1.0.59) (2021-07-22)


### üêõ„ÄÄBug Fixes

* disable some list ([f80f8be](https://github.com/bluelovers/ws-ipfs/commit/f80f8becf3b6e10bd852b2e700bd40f7fceacdf5))





## [1.0.58](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.57...ipfs-server-list@1.0.58) (2021-07-22)

**Note:** Version bump only for package ipfs-server-list





## [1.0.57](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.56...ipfs-server-list@1.0.57) (2021-07-22)


### ‚ôªÔ∏è„ÄÄChores

* **deps:** update deps ([85d6561](https://github.com/bluelovers/ws-ipfs/commit/85d6561e987b7753cdd248b73a5e74a82d4c23e7))





## [1.0.56](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.55...ipfs-server-list@1.0.56) (2021-07-21)

**Note:** Version bump only for package ipfs-server-list





## [1.0.55](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.54...ipfs-server-list@1.0.55) (2021-07-20)

**Note:** Version bump only for package ipfs-server-list





## [1.0.54](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.52...ipfs-server-list@1.0.54) (2021-07-11)


### üîñ„ÄÄMiscellaneous

* . ([823d1b4](https://github.com/bluelovers/ws-ipfs/commit/823d1b4add2fb35bc228e738708fad903ea29df1))
* . ([2742172](https://github.com/bluelovers/ws-ipfs/commit/2742172440928000c1c6aa873c933a5744c49f69))





## [1.0.53](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.52...ipfs-server-list@1.0.53) (2021-07-11)


### üîñ„ÄÄMiscellaneous

* . ([2742172](https://github.com/bluelovers/ws-ipfs/commit/2742172440928000c1c6aa873c933a5744c49f69))





## [1.0.52](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.51...ipfs-server-list@1.0.52) (2021-07-11)


### ‚ú®„ÄÄFeatures

* update server list ([490c162](https://github.com/bluelovers/ws-ipfs/commit/490c1626a46bea1b0c144d7201499703c261dadb))


### üîñ„ÄÄMiscellaneous

* . ([5631f20](https://github.com/bluelovers/ws-ipfs/commit/5631f2093178db2a887d3e92e2d233fcbb7449fb))





## [1.0.51](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.50...ipfs-server-list@1.0.51) (2021-07-11)

**Note:** Version bump only for package ipfs-server-list





## [1.0.50](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.49...ipfs-server-list@1.0.50) (2021-07-11)


### ‚ú®„ÄÄFeatures

* update server list ([46d5db6](https://github.com/bluelovers/ws-ipfs/commit/46d5db6b62fd99beebd384e1892b5b0688f9c084))





## [1.0.49](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.48...ipfs-server-list@1.0.49) (2021-07-11)


### üîñ„ÄÄMiscellaneous

* . ([f6e4d35](https://github.com/bluelovers/ws-ipfs/commit/f6e4d357a7c796c00ea915038c4b4fdb7270e9dc))





## [1.0.48](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.47...ipfs-server-list@1.0.48) (2021-07-05)

**Note:** Version bump only for package ipfs-server-list





## [1.0.47](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.46...ipfs-server-list@1.0.47) (2021-07-04)


### ‚ú®„ÄÄFeatures

* add https://ipfs.eternum.io/ipns/ ([4494bb4](https://github.com/bluelovers/ws-ipfs/commit/4494bb499cd9f22f2b836cfc41693c5c00ac0faf))


### üì¶„ÄÄCode Refactoring

* update deps and build ([a542d92](https://github.com/bluelovers/ws-ipfs/commit/a542d92420faef55f6879fedc07d563f21db03a7))





## [1.0.46](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.45...ipfs-server-list@1.0.46) (2020-08-29)


### üîñ„ÄÄMiscellaneous

* . ([a89ec77](https://github.com/bluelovers/ws-ipfs/commit/a89ec77c79a26768acfede82c769a6a792eee25b))





## [1.0.45](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.44...ipfs-server-list@1.0.45) (2020-08-29)


### üîñ„ÄÄMiscellaneous

* . ([4a550a5](https://github.com/bluelovers/ws-ipfs/commit/4a550a55ccd04d245d5935914d091a879986a8f2))





## [1.0.44](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.43...ipfs-server-list@1.0.44) (2020-08-29)


### üîñ„ÄÄMiscellaneous

* . ([cc24bf2](https://github.com/bluelovers/ws-ipfs/commit/cc24bf22e5f25f217df7c54b8671a476e5da575d))





## [1.0.43](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.42...ipfs-server-list@1.0.43) (2020-08-29)


### üîñ„ÄÄMiscellaneous

* . ([bf70f29](https://github.com/bluelovers/ws-ipfs/commit/bf70f298426c11645d5343255656fa72e0cae844))





## [1.0.42](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.41...ipfs-server-list@1.0.42) (2020-08-29)


### ‚ú®„ÄÄFeatures

* add IPNS of temporal.cloud ([5e8160c](https://github.com/bluelovers/ws-ipfs/commit/5e8160c344ba5f5e9b36fce31055fce5f74a4155))
* localServerInfo ([8fda90e](https://github.com/bluelovers/ws-ipfs/commit/8fda90eb995c382f154a411a0ffd135d6e4ce063))


### ‚ôªÔ∏è„ÄÄChores

* update deps ([7d13ceb](https://github.com/bluelovers/ws-ipfs/commit/7d13cebfde55dff62632755bfbec0af492495dc7))





## [1.0.41](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.40...ipfs-server-list@1.0.41) (2020-07-18)


### ‚ôªÔ∏è„ÄÄChores

* update deps ([f614d28](https://github.com/bluelovers/ws-ipfs/commit/f614d282b63f93b9d2a3f5755aee5d9073212bbf))





## [1.0.40](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.39...ipfs-server-list@1.0.40) (2020-07-18)

**Note:** Version bump only for package ipfs-server-list





## [1.0.39](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.38...ipfs-server-list@1.0.39) (2020-07-12)


### ‚ú®„ÄÄFeatures

* add `geesome-node` https://github.com/galtproject/geesome-node ([256fe64](https://github.com/bluelovers/ws-ipfs/commit/256fe64a3702565107341382ae44f4cea78cde9f))


### ‚ôªÔ∏è„ÄÄChores

* update deps ([40d57c8](https://github.com/bluelovers/ws-ipfs/commit/40d57c83e126c495aaf515ca8ac8cd02db848fc0))





## [1.0.38](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.37...ipfs-server-list@1.0.38) (2020-07-03)


### üîñ„ÄÄMiscellaneous

* . ([0215985](https://github.com/bluelovers/ws-ipfs/commit/02159857809e29f3a2476a54e13ab1b8a7191433))





## [1.0.37](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.36...ipfs-server-list@1.0.37) (2020-07-03)

**Note:** Version bump only for package ipfs-server-list





## [1.0.36](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.35...ipfs-server-list@1.0.36) (2020-06-16)


### ‚ôªÔ∏è„ÄÄChores

* **deps:**  update deps ([754a4c4](https://github.com/bluelovers/ws-ipfs/commit/754a4c4a714d3d256500b319473ce610f876b442))





## [1.0.35](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.34...ipfs-server-list@1.0.35) (2020-06-05)


### Features

* update deps ([a25ebd6](https://github.com/bluelovers/ws-ipfs/commit/a25ebd688ccfd54f164b3ff89cf6cdb2e7f6e478))





## [1.0.34](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.33...ipfs-server-list@1.0.34) (2020-06-05)


### Features

* update deps ([4355bc0](https://github.com/bluelovers/ws-ipfs/commit/4355bc0161fa03725b7455cee33ac834a99b7cd9))





## [1.0.33](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.32...ipfs-server-list@1.0.33) (2020-05-24)

**Note:** Version bump only for package ipfs-server-list





## [1.0.32](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.31...ipfs-server-list@1.0.32) (2020-05-17)

**Note:** Version bump only for package ipfs-server-list





## [1.0.31](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.30...ipfs-server-list@1.0.31) (2020-05-17)

**Note:** Version bump only for package ipfs-server-list





## [1.0.30](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.29...ipfs-server-list@1.0.30) (2020-05-17)


### Features

* tweak order ([959f0dd](https://github.com/bluelovers/ws-ipfs/commit/959f0ddb84d5658967c51547a764111414eb0bdd))





## [1.0.29](https://github.com/bluelovers/ws-ipfs/compare/ipfs-server-list@1.0.28...ipfs-server-list@1.0.29) (2020-05-17)


### Features

* add btfsGateway , fleek.co ([9752860](https://github.com/bluelovers/ws-ipfs/commit/9752860aed0ca25eedf0815471663fa83c283c76))
* add list from public-gateway-checker ([d9f992e](https://github.com/bluelovers/ws-ipfs/commit/d9f992e04131132dac65550014dee7e2b4a34ecf))
