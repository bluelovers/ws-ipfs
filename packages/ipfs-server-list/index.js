"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ipfsServerList = exports.filterList = exports.getIpfsLocalList = exports.getIpfsServerList = void 0;
function getIpfsServerList() {
    let data = {
        'ipfs': {
            Gateway: 'https://ipfs.io/ipfs/',
            IPLD: 'https://explore.ipld.io/#/explore/',
            IPNS: 'https://ipfs.io/ipns/',
        },
        /**
         * http://blog.hubwiz.com/2019/09/11/infura-dev-manual/
         * http://cw.hubwiz.com/card/c/infura-api/1/4/3/
         * https://github.com/Pedro-vk/ipfs-website-deployer/blob/master/src/ipfs-website-deployer-cli.ts
         * https://infura.io/docs/ipfs/get/version
         */
        'infura.io': {
            API: {
                port: 5001,
                host: 'ipfs.infura.io',
                protocol: 'https',
            },
            Gateway: 'https://ipfs.infura.io/ipfs/',
            limit: {
                ref: false,
                id: false,
                config: false,
                ls: false,
            },
        },
        /**
         * https://developers.cloudflare.com/distributed-web/ipfs-gateway/
         */
        'cloudflare': {
            Gateway: 'https://cloudflare-ipfs.com/ipfs/',
            GatewayDomain: '.cf-ipfs.com',
        },
        'ipfs.gateway': {
            Gateway: 'https://gateway.ipfs.io/ipfs/',
        },
        'bdaily': {
            Gateway: 'https://gateway.bdaily.club/ipfs/',
        },
    };
    data;
    return data;
}
exports.getIpfsServerList = getIpfsServerList;
function getIpfsLocalList() {
    let data = {
        'go-ipfs': {
            API: {
                port: 5001,
            },
        },
        'js-ipfs': {
            API: {
                port: 5002,
            },
        },
    };
    data;
    return data;
}
exports.getIpfsLocalList = getIpfsLocalList;
function filterList(key, serverList = exports.ipfsServerList) {
    return Object.keys(serverList)
        .reduce((a, b) => {
        if (serverList[b][key] != null) {
            a.push(serverList[b][key]);
        }
        return a;
    }, []);
}
exports.filterList = filterList;
exports.ipfsServerList = getIpfsServerList();
exports.default = exports.ipfsServerList;
//# sourceMappingURL=index.js.map