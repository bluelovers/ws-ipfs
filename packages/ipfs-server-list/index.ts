import { ITSRequireAtLeastOne, ITSPartialRecord } from 'ts-type'
import { IIPFSClientAddresses } from '@bluelovers/ipfs-http-client';

export type ILimit = ITSPartialRecord<'ref' | 'id' | 'config', boolean>;

export type IIPFSAddressesLike = ITSRequireAtLeastOne<{
	API?: IIPFSClientAddresses,
	Gateway?: string,
	IPLD?: string,
	IPNS?: string,
	/**
	 * https://blog.cloudflare.com/continuing-to-improve-our-ipfs-gateway/
	 *
	 * base32.cf-ipfs.com
	 */
	GatewayDomain?: string,
}> & {
	/**
	 * @deprecated
	 */
	limit?: ITSRequireAtLeastOne<ILimit>,
};

export function getIpfsServerList()
{
	let data = {
		/**
		 * http://blog.hubwiz.com/2019/09/11/infura-dev-manual/
		 * http://cw.hubwiz.com/card/c/infura-api/1/4/3/
		 * https://github.com/Pedro-vk/ipfs-website-deployer/blob/master/src/ipfs-website-deployer-cli.ts
		 * https://infura.io/docs/ipfs/get/version
		 */
		'infura.io': {
			API: {
				port: 5001,
				host: 'ipfs.infura.io',
				protocol: 'https',
			},
			Gateway: 'https://ipfs.infura.io/ipfs/',
			limit: {
				ref: false,
				id: false,
				config: false,
				ls: false,
			},
		},
		/**
		 * https://developers.cloudflare.com/distributed-web/ipfs-gateway/
		 */
		'cloudflare': {
			Gateway: 'https://cloudflare-ipfs.com/ipfs/',
			GatewayDomain: '.cf-ipfs.com',
		},
		'ipfs': {
			Gateway: 'https://ipfs.io/ipfs/',
			IPLD: 'https://explore.ipld.io/#/explore/',
			IPNS: 'https://ipfs.io/ipns/',
		},
	} as const;

	(<Record<string, IIPFSAddressesLike>>data);

	return data as Record<keyof typeof data, IIPFSAddressesLike> & Record<string, IIPFSAddressesLike>
}

export function filterList<K extends keyof IIPFSAddressesLike>(key: K, serverList: Record<string, IIPFSAddressesLike> = ipfsServerList)
{
	return Object.keys(serverList)
		.reduce((a, b) => {

			if (serverList[b][key] != null)
			{
				a.push(serverList[b][key])
			}

			return a
		}, [] as IIPFSAddressesLike[K][])
}

export const ipfsServerList = getIpfsServerList();

export default ipfsServerList
