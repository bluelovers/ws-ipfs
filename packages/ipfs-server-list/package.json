{
  "name": "ipfs-server-list",
  "version": "1.0.43",
  "description": "",
  "homepage": "https://github.com/bluelovers/ws-ipfs/tree/master/packages/ipfs-server-list#readme",
  "bugs": {
    "url": "https://github.com/bluelovers/ws-ipfs/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bluelovers/ws-ipfs.git"
  },
  "license": "ISC",
  "author": "bluelovers",
  "main": "index.js",
  "scripts": {
    "lint": "npx eslint **/*.ts",
    "test": "echo \"Error: no test specified\" && exit 1",
    "test:mocha": "npx mocha --require ts-node/register \"!(node_modules)/**/*.{test,spec}.{ts,tsx}\"",
    "npm:publish": "npm publish",
    "prepublish:lockfile": "npx sync-lockfile .",
    "prepublishOnly_": "yarn run ncu && yarn run sort-package-json && yarn run test",
    "postpublish_": "git commit -m \"chore(release): publish\" .",
    "ncu": "npx yarn-tool ncu -u",
    "sort-package-json": "npx yarn-tool sort",
    "tsc:default": "tsc -p tsconfig.json",
    "tsc:esm": "tsc -p tsconfig.esm.json"
  },
  "dependencies": {
    "@bluelovers/ipfs-http-client": "^1.0.59",
    "ipfs-types": "^1.0.28",
    "ts-type": "^1.2.32"
  },
  "gitHead": "e0a240a57b4c1a802860da8f8630d3ef9fdf3894"
}
