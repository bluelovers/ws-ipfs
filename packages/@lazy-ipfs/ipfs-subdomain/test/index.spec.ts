import ipfsSubdomain, { ipfsSubdomainURL } from '../index';
import { getIpfsServerList } from 'ipfs-server-list';

test(`ipfsSubdomain`, () =>
{
	let cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	let actual = ipfsSubdomain(cid);
	let expected;

	expect(actual).toMatchSnapshot();
	expect(() => new URL(actual)).not.toThrow();

});

test(`ipfsSubdomainURL`, () =>
{
	let cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	let actual = ipfsSubdomainURL(cid);
	let expected;

	expect(actual).toMatchSnapshot();

});

test(`ipfsSubdomain:fleek`, () =>
{
	let cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	let actual = ipfsSubdomain(cid, getIpfsServerList().fleek);
	let expected;

	expect(actual).toMatchSnapshot();
	expect(() => new URL(actual)).not.toThrow();

});
