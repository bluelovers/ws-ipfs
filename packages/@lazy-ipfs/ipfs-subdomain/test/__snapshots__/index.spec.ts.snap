// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ipfsSubdomain 1`] = `"https://bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m.ipfs.cf-ipfs.com/"`;

exports[`ipfsSubdomain:fleek 1`] = `"https://bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m.on.fleek.co/"`;

exports[`ipfsSubdomainURL 1`] = `"https://bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m.ipfs.cf-ipfs.com/"`;

exports[`ipfsSubdomainURL with path 1`] = `"https://bafybeigyzswwpp2t676ndmdxgi6qkhppugz7wdobsczod5ssas4qgsyeay.ipfs.cf-ipfs.com/dmzj"`;

exports[`ipfsSubdomainURL with path 2`] = `"https://bafybeigyzswwpp2t676ndmdxgi6qkhppugz7wdobsczod5ssas4qgsyeay.ipfs.cf-ipfs.com/dmzj"`;

exports[`ipfsSubdomainURL with path 3`] = `"https://bafybeigyzswwpp2t676ndmdxgi6qkhppugz7wdobsczod5ssas4qgsyeay.ipfs.cf-ipfs.com/"`;
